py/objint.c

```diff
@@ -363,6 +363,10 @@ mp_int_t mp_obj_int_get_checked(mp_const_obj_t self_in) {
     return MP_OBJ_SMALL_INT_VALUE(self_in);
 }
 
+mp_uint_t mp_obj_int_get_uint_checked(mp_const_obj_t self_in) {
+    return MP_OBJ_SMALL_INT_VALUE(self_in);
```

Comment:
This doesn't satisfy `Will raise exception if value doesn't fit into mp_uint_t` I think, likewise for the other implementations. So there's essentially no difference between this and having the caller write `(mp_uint_t) mp_obj_int_get_int_checked()` and that would probably also be smaller in code size. Anyway: I think this is a useful method to have, I've had to write it myself as well and it came up on the forums a couple of times, but I feel it should either be implemented properly and completely or not at all.

Reply:
You're right. This WIP commit was just taken from https://github.com/micropython/micropython/pull/4928. We should continue this discussion there and finish it properly.

Reply:
Let's continue at https://github.com/micropython/micropython/pull/4928/files#r365611830